node/src/components/consensus/era_supervisor.rs

```diff
@@ -153,6 +159,38 @@ where
         }
     }
 
+    fn highway_config(&self) -> HighwayConfig {
+        self.chainspec.genesis.highway_config
+    }
+
+    fn instance_id(&self, post_state_hash: hash::Digest, block_height: u64) -> hash::Digest {
+        let mut result = [0; hash::Digest::LENGTH];
+        let mut hasher = VarBlake2b::new(hash::Digest::LENGTH).expect("should create hasher");
+
+        hasher.input(&self.chainspec.genesis.name);
+        hasher.input(self.chainspec.genesis.timestamp.millis().to_le_bytes());
+        hasher.input(post_state_hash);
+
+        for upgrade_point in self
+            .chainspec
+            .upgrades
+            .iter()
+            .take_while(|up| up.activation_point.rank <= block_height)
+        {
+            if let Some(bytes) = upgrade_point.upgrade_installer_bytes.as_ref() {
+                hasher.input(bytes);
+            }
+            if let Some(bytes) = upgrade_point.upgrade_installer_args.as_ref() {
+                hasher.input(bytes);
+            }
```

Comment:
Maybe even hash the `activation_point` itself?

Reply:
That might be a good idea :+1: